bot/utils/messages.py

```diff
@@ -238,3 +239,102 @@ async def send_denial(ctx: Context, reason: str) -> discord.Message:
 def format_user(user: discord.abc.User) -> str:
     """Return a string for `user` which has their mention and ID."""
     return f"{user.mention} (`{user.id}`)"
+
+
+def shorten_text(text: str) -> str:
+    """
+    Truncate the text if there are over 3 lines or 300 characters, or if it is a single word.
+
+    The maximum length of the string would be 303 characters across 3 lines at maximum.
+    """
+    original_length = len(text)
+    # Truncate text to a maximum of 300 characters
+    if len(text) > 300:
+        text = text[:300]
+
+    # Limit to a maximum of three lines
+    text = "\n".join(text.split("\n", maxsplit=3)[:3])
+
+    # If it is a single word, then truncate it to 50 characters
+    if text.find(" ") == -1:
+        text = text[:50]
```

Comment:
" " not in text

Reply:
This also fails to account for nonstandard or newline delimited content. Example:

123456789012345678901234567890
123456789012345678901234567890
123456789012345678901234567890

is 60 characters, but this filter will truncate it to 50. I don't think we should do anything fancy here--it's very unlikely someone will link a message with nonbreaking spaces in it--but I think we should account for the newline case.